sdk/nodejs/tests/provider.spec.ts

```diff
@@ -495,7 +495,7 @@ describe("provider", () => {
             it(`deserializes '${test.name}' correctly`, async () => {
                 pulumi.runtime.setMocks(new TestMocks(), "project", "stack", true);
                 pulumi.runtime.registerResourceModule("test", "index", new TestModule());
-                new TestResource("name"); // Create an instance so it can be deserialized.
+                await new TestResource("name"); // Create an instance so it can be deserialized.
```

Comment:
We were previously not awaiting this resource, so when we started resolving aliases during the test because the mock supports that now, things started failing because the resource wasn't fully registered yet by the time we were checking it.  Regardless of the other changes in this PR I believe this is the right thing to do.

Reply:
AFAIK, it is not possible to `await` a constructor, since it is never `async` and definitely doesn't return a `Promise<T>` 🤔

Reply:
Await the resources URN instead.

Reply:
Interesting, somehow it seemed to work in the test.  I don't know a lot about typescript, I think I got it right now?

Reply:
Yup